### PR TITLE
WIP: Test with vtk on Travis

### PR DESCRIPTION
**Pull request by arokem**

### CODE REVIEW

**Comment by arokem** [comment]
Trying this out again...

File: .travis.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ matrix:
       env:
         - DEPENDS="cython numpy scipy matplotlib h5py nibabel cvxopt scikit_learn"
 before_install:
+    - apt-get update && apt-get install -y python-vtk
     - source tools/travis_tools.sh
     - virtualenv --python=python venv
```
Comment (matthew-brett) [inline .travis.yml]: I think you'll need `virtualenv --system-site-packages --python=python venv`

Comment (arokem) [inline .travis.yml]: Will this allow me to install with apt-get?

File: .travis.yml [hunk 2]
```diff
@@ -31,8 +31,9 @@ matrix:
       env:
         - DEPENDS="cython numpy scipy matplotlib h5py nibabel cvxopt scikit_learn"
 before_install:
+    - apt-get update && apt-get install -y python-vtk
```
Comment (matthew-brett) [inline .travis.yml]: I believe you'll need sudo for the apt-get.  I think you can use sudo for selected builds - but at the moment the global setting is `sudo: false` - which makes the builds faster.   If you want `python-vtk` for all builds, you can add it to the `apt` package list - e.g 

```
addons:
  apt:
    packages:
      - python-vtk
```

See the matplotlib `.travis.yml` file for an example.

Comment (arokem) [inline .travis.yml]: Yes - I am just reading through the docs about this.

Sadly, it seems that we will also need to ask for travis to white-list vtk:
https://github.com/travis-ci/apt-source-whitelist

Comment (matthew-brett) [inline .travis.yml]: It's a good idea to do that, but maybe using sudo for one build would be enough for now.

**Comment by matthew-brett** [comment]
Try adding a new matrix entry:

```
matrix:
   include:
     - python: 2.7
       sudo: true
       env:
         - VTK=1
```

then lower down:

```
- if [ "${VTK}" == "1" ]; then
       sudo apt-get update && apt-get install -y python-vtk;
       toggleglobalsitepackages;
       fi
```

**Comment by arokem** [comment]
Thanks! Trying that out. We'll still also need to get xvfb running here...

**Comment by arokem** [comment]
What's the `toggleglobalsitepackages;` business all about. Travis didn't seem to take to that at all.

**Comment by matthew-brett** [comment]
Ah, yes, sorry, that's a `virtualenvwrapper` feature. 

I think you'll have to do something like:

```
matrix:
   include:
     - python: 2.7
       sudo: true
       env:
         - VTK=1
         - VENV_ARGS='--system-site-packages'
```

and

```
    - virtualenv $VENV_ARGS --python=python venv
```

**Comment by arokem** [comment]
Hrrm. That's already there:

https://github.com/arokem/dipy/blob/test-vtk/.travis.yml#L39

**Comment by arokem** [comment]
Well - I can get VTK to install, and I can't get the tests to run,
seemingly with xvfb. The only remaining piece in the puzzle (I hope!) is to
get vtk to actually import within this venv. Otherwise, it's back to conda,
possibly?

**Comment by arokem** [comment]
Hey @matthew-brett - do you understand why vtk can't be imported in this build: https://travis-ci.org/arokem/dipy/jobs/85562441 ? It's based on this commit: https://github.com/arokem/dipy/commit/264348d6192b9a16393d98b9b0bdfcf336f5c905 

Am I calling virtualenvwrapper the right way there? 

It seems to run fine, but still vtk can't be imported.

**Comment by matthew-brett** [comment]
The toggle stuff is only get system site packages on the path - which should be covered by your virtualenv creation.

Why not replace everything after the virtualenv creation with `python -c 'import vtk'` to see what's going on?

**Comment by matthew-brett** [comment]
Sorry - specifically I don't think it's useful to install virtualenvwrapper - sorry for the red herring there.

**Comment by arokem** [comment]
But I do think it's about the availability of vtk to be imported.

See: https://travis-ci.org/nipy/dipy/jobs/85562784

VTK gets installed, and is immediately afterwards unavailable to import.

Maybe if we knew where this is installed, we could symlink from within the
venv, or add that to the venv PYTHONPATH?

**Comment by matthew-brett** [comment]
I see the problem - the virtualenv isn't using the system Python, which is where vtk will appear:

Failing thing showing that import vtk doesn't work : 

Succeeding thing showing that import vtk does work: https://travis-ci.org/matthew-brett/dipy/jobs/85600918

Travis.yml : https://github.com/matthew-brett/dipy/blob/c51cccb761ee4b3c07b8c8a897d67a9318003c41/.travis.yml

**Comment by arokem** [comment]
Thanks! I just pushed that change. Let's see what the golem says now.

**Comment by arokem** [comment]
Yep - it's now testing with VTK!

**Comment by Garyfallidis** [comment]
Super!

**Pull request merged**